src/Api/Resource/User.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Api\Resource;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use ApiPlatform\Core\Annotation\ApiProperty;
+
+/**
+ * @ApiResource(
+ *     collectionOperations={
+ *         "get"
+ *     },
+ *     itemOperations={
+ *         "get"
+ *     }
+ * )
+ */
+final class User
+{
+    /**
+     * @ApiProperty(identifier=true)
+     */
+    public $id;
+
+    public function __construct(string $id)
```

Comment:
It's not 100% supported right no (in write, reading is OK), but it will soon: api-platform/core#1749

Reply:
so thinking about this and the API design itself. Im leaning to dedicated write endpoints.

- /api/users
-- anonumous: sees public fields (indexed id, ...)
-- ROLE_ADMIN: sees also private fields (UUID, email, ...)
- /api/users/<indexed_id>
-- indexed id could be a short UUID maybe
- /api/users/<indexed_id>/confirm
-- allowed by ROLE_ADMIN
-- dispatched ConfirmUserCommand(<indexed_to_real_UUID>)

that what we only need read models, where write ops mostly operate on ID (could/should be simple model as well probably)

Not sure access should be tied to ROLE_ADMIN, perhaps introduce a ROLE_FULL_API_ACCESS or so?